gcloud/bigquery/job.py

```diff
@@ -919,3 +931,278 @@ def _scrub_local_properties(self, cleaned):
+                'projectId': self.project,
+                'datasetId': self.default_dataset.name,
+            }
+
+        if self.max_results is not None:
+            resource['maxResults'] = self.max_results
+
+        if self.preserve_nulls is not None:
+            resource['preserveNulls'] = self.preserve_nulls
+
+        if self.timeout_ms is not None:
+            resource['timeoutMs'] = self.timeout_ms
+
+        if self.use_query_cache is not None:
+            resource['useQueryCache'] = self.use_query_cache
+
+        return resource
+
+    def run(self, client=None):
+        """API call:  run the query via a POST request
+
+        See:
+        https://cloud.google.com/bigquery/docs/reference/v2/jobs/query
+
+        :type client: :class:`gcloud.bigquery.client.Client` or ``NoneType``
+        :param client: the client to use.  If not passed, falls back to the
+                       ``client`` stored on the current dataset.
+        """
+        client = self._require_client(client)
+        path = '/projects/%s/queries' % (self.project,)
+        api_response = client.connection.api_request(
+            method='POST', path=path, data=self._build_resource())
+        self._set_properties(api_response)
+
+    def fetch_data(self, max_results=None, page_token=None, start_index=None,
+                   timeout_ms=None, client=None):
+        """API call:  fetch a page of query result data via a GET request
+
+        See:
+        https://cloud.google.com/bigquery/docs/reference/v2/jobs/getQueryResults
+
+        :type max_results: integer or ``NoneType``
+        :param max_results: maximum number of rows to return.
+
+        :type page_token: string or ``NoneType``
+        :param page_token: token representing a cursor into the table's rows.
+
+        :type start_index: integer or ``NoneType``
+        :param start_index: zero-based index of starting row
+
+        :type timeout_ms: integer or ``NoneType``
+        :param timeout_ms: timeout, in milliseconds, to wait for query to
+                           complete
+
+        :type client: :class:`gcloud.bigquery.client.Client` or ``NoneType``
+        :param client: the client to use.  If not passed, falls back to the
+                       ``client`` stored on the current dataset.
+
+        :rtype: tuple
+        :returns: ``(row_data, total_rows, page_token)``, where ``row_data``
+                  is a list of tuples, one per result row, containing only
+                  the values;  ``total_rows`` is a count of the total number
+                  of rows in the table;  and ``page_token`` is an opaque
+                  string which can be used to fetch the next batch of rows
+                  (``None`` if no further batches can be fetched).
+        :raises: ValueError if the query has not yet been executed.
+        """
+        if self.name is None:
+            raise ValueError("Query not yet executed:  call 'run()'")
```

Comment:
So if a user wanted to read from that temporary table would they use `RunSyncQueryJob.fetch_data` or something else?